docs/DeveloperGuide.adoc

```diff
@@ -88,61 +106,78 @@ The `UI` component,
 * Executes user commands using the `Logic` component.
 * Listens for changes to `Model` data so that the UI can be updated with the modified data.
 
+Given below is the Sequence Diagram for interactions within
+the UI component when the user enter an add command.
+The exact command entered is `add t/Animal Farm a/George`.
+
+.Interactions Inside the UI Component for the `add t/Animal Farm a/George` Command
+image::UiAddBookSequenceDiagram.png[]
+{nbsp} +
+
+The following activity diagram summarizes what happens to the `UI`
+when a user executes a new command:
+
+.Flow of Events within `UI`
+image::UiUpdateBookListActivityDiagram.png[width=270]
+{nbsp} +
+
 [[Design-Logic]]
 === Logic component
 
 [[fig-LogicClassDiagram]]
 .Structure of the Logic Component
 image::LogicClassDiagram.png[]
+{nbsp} +
 
 *API* :
 link:{repoURL}/src/main/java/seedu/address/logic/Logic.java[`Logic.java`]
 
-.  `Logic` uses the `AddressBookParser` class to parse the user command.
+.  `Logic` uses the `CatalogParser` class to parse the user command.
 .  This results in a `Command` object which is executed by the `LogicManager`.
-.  The command execution can affect the `Model` (e.g. adding a person).
+.  The command execution can affect the `Model` (e.g. adding a book).
 .  The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
 .  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
 
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("add t/Animal Farm a/George")` API call.
 
-.Interactions Inside the Logic Component for the `delete 1` Command
-image::DeleteSequenceDiagram.png[]
+.Interactions Inside the Logic Component for the `add t/Animal Farm a/George` Command
+image::AddBookLogicSequenceDiagram.png[]
+{nbsp} +
 
-NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+NOTE: The lifeline for `AddCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
 image::ModelClassDiagram.png[]
```

Comment:
Perhaps the multiplicity for Tag could be included too, for consistency.

Reply:
The multiplicity is included in the diagram, but it may not be obvious since it is blocked by the `Tag` title. Will try to use hidden arrows to expose it.